modules/misc/qt.nix

```diff
@@ -173,26 +175,27 @@ in {
       QT_STYLE_OVERRIDE = cfg.style.name;
     };
 
-    home.packages = if cfg.platformTheme == "gnome" then
+    home.packages = (if cfg.platformTheme == "gnome" then
       [ pkgs.qgnomeplatform ]
-      ++ lib.optionals (cfg.style.package != null) [ cfg.style.package ]
     else if cfg.platformTheme == "qtct" then [
       pkgs.libsForQt5.qt5ct
       pkgs.qt6Packages.qt6ct
     ] else if cfg.platformTheme == "kde" then [
       pkgs.libsForQt5.plasma-integration
       pkgs.libsForQt5.systemsettings
     ] else
-      [ pkgs.libsForQt5.qtstyleplugins ];
+      [ pkgs.libsForQt5.qtstyleplugins ])
+      ++ lib.optionals (cfg.style.package != null)
+      (lib.flatten [ cfg.style.package ]);
 
     xsession.importedVariables = [ "QT_QPA_PLATFORMTHEME" ]
       ++ lib.optionals (cfg.style.name != null) [ "QT_STYLE_OVERRIDE" ];
 
-    # Enable GTK+ style for Qt4 in either case.
+    # Enable GTK+ style for Qt4 in Gtk/GNOME.
     # It doesn’t support the platform theme packages.
-    home.activation.useGtkThemeInQt4 = hm.dag.entryAfter [ "writeBoundary" ] ''
-      $DRY_RUN_CMD ${pkgs.crudini}/bin/crudini $VERBOSE_ARG \
-        --set "${config.xdg.configHome}/Trolltech.conf" Qt style GTK+
-    '';
```

Comment:
This Crudini project only exists to manipulate INI files: https://github.com/pixelb/crudini

We have way better options now to manage those in Nix.

Reply:
We could perhaps condition this on `stateVersion`.

Reply:
> This might break existing configurations. Shouldn't Trolltech.conf stay writable? What other applications might modify it?
I never saw other applications writing this file.

Reply:
@ncfavier Changed my mind. Will keep the activation script because while I don't think this file is written by other applications, I am not 100% sure and I just realize I don't have any Qt4 applications to test.

So for now the only change I did is to only apply this file when `qt.platformTheme` is either `gtk` or `gnome`.